src/components/GraphiQL.js

```diff
@@ -167,7 +167,17 @@ export class GraphiQL extends React.Component {
     if (nextSchema !== this.state.schema ||
         nextQuery !== this.state.query ||
         nextOperationName !== this.state.operationName) {
-      this._updateQueryFacts(nextQuery);
+      const updatedQueryAttributes = this._updateQueryFacts(
+        nextQuery,
+        nextOperationName,
+        nextSchema,
+      );
+
+      if (updatedQueryAttributes !== undefined) {
+        nextOperationName = updatedQueryAttributes.operationName;
+
+        this.setState({ ...updatedQueryAttributes.queryFacts });
```

Comment:
shouldn't this be `this.setState(updatedQueryAttributes)`? `.queryFacts` isn't a property set anywhere, the variable by that name is spread into the return value